Assets/Tests/EditMode/AuthTests/TokenSaving/AuthTokenSaverTests.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using NUnit.Framework;
+using Src.Auth;
+using Src.Auth.AuthTokenSaver;
+using Src.Auth.JwtManagement;
+using Src.Auth.JwtManagement.Tokens;
+using Tests.Utils.Mocks;
+using UnityEngine;
+
+namespace Tests.EditMode.AuthTests.TokenSaving
+{
+    [TestFixture]
+    public class AuthTokenSaverTests
+    {
+        private const string SampleToken = "SAMPLE_TOKEN";
+        private const string LastLoginKey = "LastLoginAuthTokensStoreInfo";
+
+        private AuthTokenSaver _tokenSaver;
+        private StringSaverMock _saverMock;
+        private AbstractJwtStore _sampleStore;
+
+        [SetUp]
+        public void Init()
+        {
+            _saverMock = new StringSaverMock();
+            _tokenSaver = new AuthTokenSaver(_saverMock);
+            
+            var sampleToken = new JwtToken(SampleToken, int.MaxValue, DateTime.Now);
+            _sampleStore = new JwtStore(sampleToken, sampleToken);
+        }
+
+
+        [Test]
+        [TestCaseSource(nameof(GetAuthTypes))]
+        public void SaveAuthTokens_ShouldSaveTokens_WithSpecifiedName(AuthType type)
+        {
+            _tokenSaver.SaveAuthTokens(_sampleStore, type);
+
+            var stored = _saverMock.Values[AuthTokenSaver.GetStorePrefNameByAuthType(type)];
+            var serialized = JsonConvert.SerializeObject(_sampleStore);
+            
+            Assert.IsTrue(stored == serialized);
+        }
+
+
+        [Test]
+        [TestCaseSource(nameof(GetAuthTypes))]
+        public void TryGetTokenValueByAuthType_ShouldReturnStoredToken_IfThereIsSavedOne(AuthType type)
+        {
+            _tokenSaver.SaveAuthTokens(_sampleStore, type);
+            _tokenSaver.TryGetTokenStoreByAuthType(out var store, type);
+            Assert.IsTrue(_sampleStore.Equals(store));
+        }
+
+
+        [Test]
+        [TestCaseSource(nameof(GetAuthTypes))]
+        public void UpdateLastLoginInfo_ShouldSaveGivenInfo(AuthType type)
+        {
+            _tokenSaver.UpdateLastLoginInfo(type);
+            var storedValue = _saverMock.Values[LastLoginKey];
+            _saverMock.Values.Remove(LastLoginKey);
```

Comment:
nit: I would create new instance of saver mock for each test instead of having one for every test. Having one _saverMock shared by all test may lead to multiple fails of tests in case if someone will forget to write _saverMock.Values.Remove(*whatever*).

Reply:
Remove method isn't nesessary to call after a test is completed. It tend to be called to emulate different internal TokenSaver state changes inside tests' scenarios, but with every new test [SetUp]-method creates new one object of this type.